crates/sound/src/audio_backends/pw_backend.rs

```diff
@@ -22,17 +25,80 @@ pub struct PCMParams {
     pub rate: u8,
 }
 
-// SAFETY: Safe as the structure can be sent to another thread.
-unsafe impl Send for WrapMainLoop {}
+#[derive(Clone)]
+pub struct PwThreadLoop(NonNull<pw_thread_loop>);
+
+impl PwThreadLoop {
+
+    pub fn new(name: Option<&str>) -> Option<Self> {
+        unsafe {
```

Comment:
I would suggest to user unsafe blocks as small as possible. E.g here maybe only `pw_thread_loop_new` needs it.

So here for example we can do:
```diff
     pub fn new(name: Option<&str>) -> Option<Self> {
-        unsafe {
-            let inner = pw_thread_loop_new(
+        let inner = unsafe {
+            pw_thread_loop_new(
                 name.map_or(ptr::null(), |p| p.as_ptr() as *const _),
                 std::ptr::null_mut(),
-            );
-            if inner.is_null() {
-                None
-            } else {
-                Some(Self(NonNull::new(inner).expect("pw_thread_loop can't be null")))
-            }
+            )
+        };
+        if inner.is_null() {
+            None
+        } else {
+            Some(Self(
+                NonNull::new(inner).expect("pw_thread_loop can't be null"),
+            ))
         }
     }
```